llvm/lib/CodeGen/MachineLICM.cpp

```diff
@@ -123,6 +123,7 @@ namespace {
     const TargetRegisterInfo *TRI = nullptr;
     const MachineFrameInfo *MFI = nullptr;
     MachineRegisterInfo *MRI = nullptr;
+    RegisterClassInfo RegClassInfo;
```

Comment:
The way RegisterClassInfo is currently used, it gets recomputed for every pass that uses it. It should probably move to be a normal analysis

Reply:
I was thinking this, especially as it has some saved state to work out when it needs to recompute. I think that's probably a good follow-up?

Reply:
Relatedly, we also do not serialize the set of reserved registers in MIR